packages/lib/services/searchengine/queryBuilder.ts

```diff
@@ -321,44 +331,61 @@ const dateFilter = (terms: Term[], conditons: string[], params: string[], relati
+	genericFilter(unixDateTerms, conditons, params, relation, 'date', useFTS);
 };
 
-const sourceUrlFilter = (terms: Term[], conditons: string[], params: string[], relation: Relation) => {
+const sourceUrlFilter = (terms: Term[], conditons: string[], params: string[], relation: Relation, useFTS: boolean) => {
 	const urlTerms = terms.filter(x => x.name === 'sourceurl');
-	genericFilter(urlTerms, conditons, params, relation, 'sourceurl');
+	genericFilter(urlTerms, conditons, params, relation, 'sourceurl', useFTS);
 };
 
+const trimQuotes = (str: string) => str.startsWith('"') && str.endsWith('"') ? str.substr(1, str.length - 2) : str;
 
-const textFilter = (terms: Term[], conditions: string[], params: string[], relation: Relation) => {
-	const addExcludeTextConditions = (excludedTerms: Term[], conditions: string[], params: string[], relation: Relation) => {
-		const type = excludedTerms[0].name === 'text' ? '' : `.${excludedTerms[0].name}`;
+const textFilter = (terms: Term[], conditions: string[], params: string[], relation: Relation, useFTS: boolean) => {
+	const createLikeMatch = (term: Term, negate: boolean) => {
+		const query = `${relation} ${negate ? 'NOT' : ''} (
+			${(term.name === 'text' || term.name === 'body') ? 'notes.body LIKE ? ' : ''}
+			${term.name === 'text' ? 'OR' : ''}
+			${(term.name === 'text' || term.name === 'title') ? 'notes.title LIKE ? ' : ''})`;
 
-		if (relation === 'AND') {
-			conditions.push(`
-			AND ROWID NOT IN (
-				SELECT ROWID
-				FROM notes_fts
-				WHERE notes_fts${type} MATCH ?
-			)`);
-			params.push(excludedTerms.map(x => x.value).join(' OR '));
-		}
+		conditions.push(query);
+		const param = `%${trimQuotes(term.value).replace(/\*/, '%')}%`;
+		params.push(param);
+		if (term.name === 'text') params.push(param);
+	};
 
-		if (relation === 'OR') {
-			excludedTerms.forEach(term => {
+	const addExcludeTextConditions = (excludedTerms: Term[], conditions: string[], params: string[], relation: Relation) => {
+		if (useFTS) {
+			const type = excludedTerms[0].name === 'text' ? '' : `.${excludedTerms[0].name}`;
+			if (relation === 'AND') {
 				conditions.push(`
-				OR ROWID IN (
-					SELECT *
-					FROM (
-						SELECT ROWID
-						FROM notes_fts
-						EXCEPT
-						SELECT ROWID
-						FROM notes_fts
-						WHERE notes_fts${type} MATCH ?
-					)
-				)`);
-				params.push(term.value);
+												AND ROWID NOT IN (
+													SELECT ROWID
+													FROM notes_fts
+													WHERE notes_fts${type} MATCH ?
+												)`);
+				params.push(excludedTerms.map(x => x.value).join(' OR '));
+			}
+			if (relation === 'OR') {
+				excludedTerms.forEach(term => {
+					conditions.push(`
+													OR ROWID IN (
+														SELECT *
+															FROM (
+																SELECT ROWID
+																FROM notes_fts
+																EXCEPT
+																SELECT ROWID
+																FROM notes_fts
+																WHERE notes_fts${type} MATCH ?
+														)
+													)`);
```

Comment:
Not sure if it's the diff doing this but could you check these strings and make sure they are indented properly?

Reply:
That's weird, it was looking way better in my editor. I did remove some indentation though to make it uniform with the other strings in the file.